src/clj_momo/lib/clj_time/core.cljc

```diff
@@ -0,0 +1,614 @@
+     (last-day-of-the-month- [this]
+       (time-delegate/last-day-of-the-month- this))
+     (week-number-of-year [this]
+       (time-delegate/week-number-of-year this))
+
+     goog.date.DateTime
+     (year [this] (time-delegate/year this))
+     (month [this] (time-delegate/month this))
+     (day [this] (time-delegate/day this))
+     (day-of-week [this] (time-delegate/day-of-week this))
+     (hour [this] (time-delegate/hour this))
+     (minute [this] (time-delegate/minute this))
+     (second [this] (time-delegate/second this))
+     (milli [this] (time-delegate/milli this))
+     (equal? [this that]
+       (when that
+         (time-delegate/equal? this that)))
+     (after? [this that]
+       (time-delegate/after? this that))
+     (before? [this that]
+       (time-delegate/before? this that))
+     (plus- [this period]
+       (time-delegate/plus- this period))
+     (minus- [this period]
+       (time-delegate/minus- this period))
+     (first-day-of-the-month- [this]
+       (time-delegate/first-day-of-the-month- this))
+     (last-day-of-the-month- [this]
+       (time-delegate/last-day-of-the-month- this))
+     (week-number-of-year [this]
+       (time-delegate/week-number-of-year this))
+
+     goog.date.Date
+     (year [this] (time-delegate/year this))
+     (month [this] (time-delegate/month this))
+     (day [this] (time-delegate/day this))
+     (day-of-week [this] (time-delegate/day-of-week this))
+     (hour [this] (time-delegate/hour this))
+     (minute [this] (time-delegate/minute this))
+     (second [this] (time-delegate/second this))
+     (milli [this] (time-delegate/milli this))
+     (equal? [this that]
+       (when that
+         (time-delegate/equal? this that)))
+     (after? [this that]
+       (time-delegate/after? this that))
+     (before? [this that]
+       (time-delegate/before? this that))
+     (plus-[this period]
+       (time-delegate/plus- this period))
+     (minus- [this period]
+       (time-delegate/minus- this period))
+     (first-day-of-the-month- [this]
+       (time-delegate/first-day-of-the-month- this))
+     (last-day-of-the-month- [this]
+       (time-delegate/last-day-of-the-month- this))
+     (week-number-of-year [this]
+       (time-delegate/week-number-of-year this))))
+
+(defn internal-now []
+  #?(:clj (Date.)
+     :cljs (now)))
+
+(defn date-time
+  ([year]
+   (time-delegate/date-time year))
+  ([year month]
+   (time-delegate/date-time year month))
+  ([year month day]
+   (time-delegate/date-time year month day))
+  ([year month day hour]
+   (time-delegate/date-time year month day hour))
+  ([year month day hour minute]
+   (time-delegate/date-time year month day hour minute))
+  ([year month day hour minute second]
+   (time-delegate/date-time year month day hour minute second))
+  ([year month day hour minute second millis]
+   (time-delegate/date-time year month day hour minute second millis)))
+
+#?(:clj
+   (defn internal-date
+     ([year]
```

Comment:
Since these functions duplicate each other, I suggest passing one to the next, and only the most specific doing the full operation. Can break it into groups of functions, depending on the underlying functions to be called:
```clojure
([year] (internal-date year 1 1))
([year month] (internal-date year month 1)
([year month day]
 (.getTime (doto (GregorianCalendar. year (dec month) day)
             (.setTimeZone TimeZoneUTC))))
([year month day hour] (internal-date year month hour 0 0))
([year month day hour minute] (internal-date year month hour minute 0))
([year month day hour minute second]
 (.getTime (doto (GregorianCalendar. year (dec month) day hour minute second)
                  (.setTimeZone TimeZoneUTC))))
([year month day hour minute second millis]
 (.getTime (doto (GregorianCalendar. year (dec month) day hour minute second)
                (.setTimeZone TimeZoneUTC)
                (.set Calendar/MILLISECOND millis)))))
```